### ISSUE TITLE
OpenGL ES support?

### ISSUE THREAD

**Issue by DonFlymoor** [pioneerspacesim/pioneer#4892]
When built from source on a Raspberry Pi 4, OpenGL fails to initiate with this message:
```
ver 20200620 on: Linux

System Name: Linux
Host Name: Pi4
Release(Kernel) Version: 5.4.47-v8+
Kernel Build Timestamp: #1322 SMP PREEMPT Wed Jun 17 18:05:46 BST 2020
Machine Arch: aarch64
Domain Name: (none)

--------------------
SDL Version (build) 2.0.9
SDL Version (dynamic) 2.0.9
SDL Versions match
SDL_image Version (build): 2.0.4
SDL_image Version (dynamic): 2.0.4
SDL_image Versions match
Assimp Version: 4.1.0
FreeType Version: 2.9.1
GLEW dynamic version: 2.0.0
--------------------

Failed to set video mode. (Couldn't find matching GLX visual). Re-trying without multisampling.
Failed to set video mode. (Couldn't find matching GLX visual). Re-trying with 16-bit depth buffer
Failed to set video mode. (Couldn't find matching GLX visual). Re-trying with 16-bit depth buffer and no multisampling
Warning: Failed to set video mode: Couldn't find matching GLX visualFatal: Failed to set video mode: No message system available
```
Upon further investigation, `glxinfo` says that my OpenGL version is 2.1, while my OpenGL ES version is 3.1.

Pioneer currently requires OpenGL 3.1, 3.2, or 3.3, which I don't have.

According to [this post](https://stackoverflow.com/questions/4519264/opengl-es-versus-opengl#:~:text=The%20main%20difference%20between%20the,(using%20a%20compatibility%20profile).), the only difference in coding is the `glBegin()` and `glEnd()` fixed pipeline coding, which is superseded by VBO's. I believe that [this](https://www.khronos.org/opengles/) website has some more information about GLES.

Is GLES support a planned feature for pioneer?

__My pioneer version (and OS):__
ver 20200620 on: ARM Linux

**Comment by impaktor** [comment]
https://pioneerspacesim.net/forum/viewtopic.php?f=3&t=242

**Comment by DonFlymoor** [comment]
Does Pioneer support OpenGL 2.1? That's the only thing stopping it from working.

**Comment by impaktor** [comment]
No, 2.1 support was removed some year(s) ago.

**Comment by DonFlymoor** [comment]
That makes sense, OpenGL 2.1 is somewhat outdated. Are there any planes on adding GLES support? From what I can find, it wouldn't take much, unless you rely on fixed pipelines.

**Comment by Web-eWorks** [comment]
There are a number of issues with supporting GLES, the largest one of which is simply that GLES uses a *different shader language* compared to regular OpenGL (specifically, variable declarations, shader linking semantics, etc. all differ between GLES and regular OpenGL, in addition to the vastly differing levels of driver support for ES vs mainstream OpenGL). The work we would have to do to support GLES in addition to regular OpenGL (which we're working in the background on deprecating and moving to a minimal-features-required Vulkan-based rendering engine) is simply non-feasible.

Simply put, we don't have any plans to support GLES now, or at any point in the future. If Raspberry Pi gets a Vulkan driver you may well eventually be able to run Pioneer there, but don't get your hopes too high... Pioneer has performance issues on regular Intel integrated graphics, and I doubt an RPI will be able to run it with any better performance.
